plugins/default/lib/component_health.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+import nagios
+import sys
+import json
+import argparse
+import urllib2
+from subprocess import CalledProcessError
+
+class RequestError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-H","--host", action="store", required=True, help="Host")
+    parser.add_argument('-P','--port', action="store", required=True, help="Port")
+    parser.add_argument("-E","--endpoint", action="store", required=False, help="Endpoint", default="/sys/info/health")
+    return parser
+
+def do_request(host, port, endpoint):
+    url =  'http://%s:%s%s' % (host,port,endpoint)
+    try:
+        data = urllib2.urlopen(url).read()
+    except urllib2.HTTPError, error:
+        if error.code == 500:
+            # For some reason the health endpoint returns a http status code of 500 when the health isn't ok, so we have to check the http error for a health response here
+            try:
+                data = error.read()
+                json.loads(data)
+            except ValueError:
+                raise RequestError(error)
+        else:
+            raise RequestError(error)
+    except urllib2.URLError:
+        raise RequestError('The health endpoint at "%s" is not contactable. This could indicate that the service is down or the url is invalid!' % url)
+    return data
+
+def parse_response(response):
+    data = json.loads(response)
+    if data['status'] == 'ok':
+        nagios_status = nagios.OK
+    elif data['status'] == 'warn':
+        nagios_status = nagios.WARN
+    else:
+        nagios_status = nagios.CRIT
+    return (nagios_status, data['summary'], data['details'])
+
+def report(summary, test_results):
+    print summary
+    for test in test_results:
+        print 'Test: %s - Status: %s - Details: %s' % (test['description'], test['test_status'],test['result'])
+
+def main():
+    parser = generate_parser().parse_args()
+    host = parser.host
+    port = parser.port
+    endpoint = parser.endpoint
+    try:
+        response = do_request(host, port, endpoint)
+    except RequestError, error:
+        print error.message
+        sys.exit(nagios.CRIT)
+    nagios_status, test_summary, test_results = parse_response(response)
+    report(test_summary, test_results)
+    sys.exit(nagios_status)
+
+if __name__ == "__main__":
+    try:
+        main()
+    except CalledProcessError:
+        pass
+    except KeyError:
+        pass
+    except Exception:
```

Comment:
I had something similar to this at one point, and then didn't like the fact that main was returning a value for some reason and thought it was thing that should be dealing with the sys.exit.